### PR TITLE
Made a few changes to Redstack script to increase functionality with Vagrant / Virtualbox.

### PR DESCRIPTION
**Pull request by TimSimpsonR**
- Put /integration/scripts into an environment variable, since some parts didn't work without a full path.
- Added function to all bash functions for consistency and clarity.
- Added code to install Reddwarf Client (we were checking the source out but not installing it).
- Added some extra devstack arguments which seem to have made networking function in VirtualBox.
- Added a set-image command.
- Added a debug command which turns on trace.
- Upped the memory of the Vagrant Box.

### CODE REVIEW

File: scripts/create_vm [hunk 1]
```diff
@@ -60,6 +60,9 @@ Vagrant::Config.run do |global_config|
     config.vm.host_name = "host"
 
     config.ssh.timeout  = 3600
+    config.vm.customize do |vm|
+        vm.memory_size = 2048
```
Comment (hub-cap) [inline scripts/create_vm:64]: Very nice :D

File: scripts/redstack [hunk 2]
```diff
@@ -311,19 +329,34 @@ run_review_for() {
...
+# to force lock_path to /tmp.
+EXTRA_OPTS=(lock_path=/tmp)
+# This is similar to code found at
+# https://github.com/bcwaldon/vagrant_devstack/blob/master/Vagrantfile
+# and seems to make instances ping'able in VirtualBox.
+FLAT_INTERFACE=eth1
+PUBLIC_INTERFACE=eth1
+FLOATING_RANGE=`ip_chunk eth0 1`.`ip_chunk eth0 2`.`ip_chunk eth0 3`.128/28
+HOST_IP=`ip_chunk eth0 1`.`ip_chunk eth0 2`.`ip_chunk eth0 3`.`ip_chunk eth0 4`
+" > localrc
```
Comment (hub-cap) [inline scripts/redstack]: Im scared of all this.

File: scripts/redstack [hunk 3]
```diff
@@ -492,23 +530,23 @@ function build_vm() {
...
     # PATH_REDDWARF more than likely has file separators, which sed does not like
     # This will escape them
     ESCAPED_PATH_REDDWARF=`echo $PATH_REDDWARF | sed 's/\//\\\\\//g'`
     sed -i.bak "s/PATH_REDDWARF/$ESCAPED_PATH_REDDWARF/g" /tmp/bootstrap_init.sh
     sed -i.bak "s/USERNAME/$USERNAME/g" /tmp/bootstrap_init.sh
-    
+
     # Apt repo on the local machine
-    cp files/apt_repo.list /tmp
+    cp $REDSTACK_SCRIPTS/files/apt_repo.list /tmp
```
Comment (hub-cap) [inline scripts/redstack:551]: i like the addition here!

File: scripts/redstack [hunk 4]
```diff
@@ -793,37 +853,51 @@ print_usage() {
...
+        "nova-client" ) shift; cmd_nova_client $@;;
+        "rd-client" ) shift; cmd_rd_client $@;;
+        "nova-delete" ) shift; cmd_delete_nova_instance $@;;
+        "wipe-logs" ) cmd_wipe_logs;;
+        "rd-sql" ) shift; cmd_rd_sql $@;;
+        "run-ci" ) shift; cmd_run_ci $@;;
+        "vagrant-ssh" ) shift; cmd_vagrant_ssh $@;;
+        "debug" ) shift; echo "Enabling debugging."; \
+                  set -o xtrace; run_command $@;;
+        "set-image" ) shift; cmd_set_image $@;;
```
Comment (hub-cap) [inline scripts/redstack:916]: VERY AWESOME

**Comment by hub-cap** [comment]
only 1 thing scares me, the eth1 stuff, and all those localrc adds. But overall, good work. I LOVE the add of the word function everywhere.

**Comment by TimSimpsonR** [comment]
Maybe we can add another stack.rc like-file that gets generated along with the create _vm script that we'd read in at the start of redstack to drive things like the local rc additions.

File: scripts/redstack [hunk 5]
```diff
@@ -492,23 +539,23 @@ function build_vm() {
     # and executes the guestagent as a service. New code to reddwarf does not require
     # a new image, just a new create of an instance.
     # Please note, this is ONLY for dev and should not be considered production ready.
-    cp files/bootstrap_init.sh /tmp
+    cp $REDSTACK_SCRIPTS/files/bootstrap_init.sh /tmp
```
Comment (cp16net) [inline scripts/redstack:542]: this is what i need!

File: scripts/redstack [hunk 6]
```diff
@@ -754,6 +801,28 @@ function cmd_rd_sql() {
...
+
+
+function cmd_run_ci() {
+    exclaim "Running CI suite..."
+    cmd_install
+    cmd_initialize
+    cmd_build_image
+    cmd_stop_deps
+    cmd_start_deps &
+    cmd_int_tests
```
Comment (cp16net) [inline scripts/redstack]: we also need to `cmd_start` before running the integration tests

**Comment by TimSimpsonR** [comment]
Made the eth0 stuff optional.

File: scripts/redstack [hunk 7]
```diff
@@ -311,19 +333,39 @@ run_review_for() {
...
+        echo "
+# This is similar to code found at
+# https://github.com/bcwaldon/vagrant_devstack/blob/master/Vagrantfile
+# and seems to make instances ping'able in VirtualBox.
+FLAT_INTERFACE=eth1
+PUBLIC_INTERFACE=eth1
+FLOATING_RANGE=`ip_chunk eth0 1`.`ip_chunk eth0 2`.`ip_chunk eth0 3`.128/28
+HOST_IP=`ip_chunk eth0 1`.`ip_chunk eth0 2`.`ip_chunk eth0 3`.`ip_chunk eth0 4`
+" >> localrc
+    fi
```
Comment (hub-cap) [inline scripts/redstack:364]: Im not so scared anymore!

File: scripts/redstack (hunk 6 above)
Comment (TimSimpsonR) [inline scripts/redstack]: Fixed.

**Comment by TimSimpsonR** [comment]
Added a clear command.

**Comment by cp16net** [comment]
+1

**Pull request merged**